### ISSUE TITLE
`Style/ParenthesesAroundCondition` autocorrects code into invalid Ruby

### ISSUE THREAD

**Issue by a-lavis** [rubocop/rubocop#12529]
`Style/ParenthesesAroundCondition` can sometimes autocorrect code into invalid Ruby.
This seems to happen with multiline `while` statements.

--------

## Expected behavior

RuboCop should not autocorrect in this case.  Maybe it should also ignore this kind of scenario?

## Actual behavior

It autocorrects this case, and thus produces invalid Ruby (which RuboCop cannot parse).

When running `rubocop -a --debug test.rb` (see the below example from the "Steps to reproduce the problem" section), I see:
```
For /Users/aidanlavis: Default configuration from /Users/aidanlavis/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-1.58.0/config/default.yml
Use parallel by default.
Skipping parallel inspection: only a single file needs inspection
Inspecting 1 file
Scanning /Users/aidanlavis/test.rb
Loading cache from /Users/aidanlavis/.cache/rubocop_cache/e41c51a0320816d8293460f2a4dbe0390fc1f056/6d7a3b621ca1730e04accd938619e4bdab66cfb1/abbd0b365a1eddc3b393f5056e799fe44f87a49f
F

Offenses:

test.rb:3:7: C: [Corrected] Style/ParenthesesAroundCondition: Don't use parentheses around the condition of a while.
while (0.upto((rand * 10).ceil).reduce(true) do |try_again, num| ...
      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
test.rb:3:48: F: Lint/Syntax: unexpected token tPIPE
(Using Ruby 2.7 parser; configure using TargetRubyVersion parameter, under AllCops)
while 0.upto((rand * 10).ceil).reduce(true) do |try_again, num|
                                               ^
test.rb:3:63: F: Lint/Syntax: unexpected token tPIPE
(Using Ruby 2.7 parser; configure using TargetRubyVersion parameter, under AllCops)
while 0.upto((rand * 10).ceil).reduce(true) do |try_again, num|
                                                              ^
test.rb:7:1: F: Lint/Syntax: unexpected token kEND
(Using Ruby 2.7 parser; configure using TargetRubyVersion parameter, under AllCops)
end
^^^

1 file inspected, 4 offenses detected, 1 offense corrected
Finished in 0.09241699999984121 seconds
```

## Steps to reproduce the problem

Create a file `test.rb`
```ruby
# frozen_string_literal: true

while (0.upto((rand * 10).ceil).reduce(true) do |try_again, num|
         puts num
         try_again && num < 8
       end)
end
```

You can run this file `ruby test.rb` and nothing breaks.
If you run `rubocop -a test.rb`, it autocorrects the file to:
```ruby
# frozen_string_literal: true

while 0.upto((rand * 10).ceil).reduce(true) do |try_again, num|
         puts num
         try_again && num < 8
       end
end
```

This file cannot be parsed - if you try to run it by doing `ruby test.rb` or if you try to run RuboCop on it (`rubocop test.rb`) it will fail.

## RuboCop version

```bash
> rubocop -V
1.58.0 (using Parser 3.2.2.4, rubocop-ast 1.30.0, running on ruby 3.2.2) [arm64-darwin22]
```
